src/mango.ts

```diff
@@ -0,0 +1,493 @@
+        Serum3OrderType.limit,
+        deltaOrderId,
+        10,
+      );
+    } else {
+      const perpHedgeSide = hedgeSide === HedgeSide.buy ? PerpOrderSide.bid : PerpOrderSide.ask;
+      await mangoClient.perpPlaceOrder(
+        mangoGroup,
+        mangoAccount,
+        perpMarket.perpMarketIndex,
+        perpHedgeSide,
+        hedgePrice,
+        Math.abs(hedgeDeltaClip),
+        undefined,
+        deltaOrderId,
+        PerpOrderType.limit,
+      );
+    }
+  } catch (err) {
+    console.log(scalper.symbol, 'Failed to place order', err, err.stack);
+  }
+
+  // Wait the twapInterval of time to see if the position gets to neutral.
+  console.log(scalper.symbol, 'Scan Delta Fills for ~', TWAP_INTERVAL_SEC, 'seconds');
+  await waitForFill(
+    (_) => Math.abs(hedgeDeltaTotal) < (deltaThreshold),
+    TWAP_INTERVAL_SEC,
+  );
+
+  // Cleanup listener.
+  fillFeed.removeEventListener('message', deltaFillListener);
+
+  if (Math.abs(hedgeDeltaTotal) < (deltaThreshold)) {
+    fillFeed.removeEventListener('message', deltaFillListener);
+    console.log(scalper.symbol, 'Delta Hedge Complete', hedgeDeltaTotal, '<', deltaThreshold);
+    return;
+  }
+
+  // Recursive call, happens when we are not getting fills to get to neutral.
+  await deltaHedgeMango(
+    dipProduct,
+    scalper,
+    mangoClient,
+    mangoAccount,
+    mangoGroup,
+    perpMarket,
+    spotMarket,
+    fillFeed,
+    deltaHedgeCount + 1,
+  );
+}
+
+export async function gammaScalpMango(
+  dipProduct: DIPDeposit[],
+  scalper: Scalper,
+  mangoClient: MangoClient,
+  mangoAccount: MangoAccount,
+  mangoGroup: Group,
+  perpMarket: PerpMarket,
+  fillFeed: WebSocket,
+  gammaScalpCount: number,
+): Promise<void> {
+  await mangoAccount.reload(mangoClient);
+  await mangoGroup.reloadAll(mangoClient);
+  // Makes the recursive gamma scalps safer. Rerun will clear any stale
+  // orders. Allows only 2 gamma orders at any time
+  await cancelStaleMangoOrders(scalper, mangoClient, mangoAccount, mangoGroup, perpMarket);
+
+  // Avoid unsafe recursion.
+  if (gammaScalpCount > GAMMA_CYCLES) {
+    console.log(scalper.symbol, 'Maximum scalps acheived!', gammaScalpCount - 1, 'Wait for Rerun');
+    return;
+  }
+
+  const fairValue = perpMarket.uiPrice;
+  const dipTotalGamma = getDIPGamma(dipProduct, fairValue, scalper.symbol);
+
+  // TODO: Allow scalper modes for back bids & strike adjustments
+  // Calc scalperWindow std deviation spread from zScore & IV for gamma levels
+  const stdDevSpread = (scalper.impliedVol
+    / Math.sqrt(SEC_PER_YEAR / SCALPER_WINDOW_SEC))
+    * scalper.zScore;
+  const netGamma = IS_DEV
+    ? Math.max(0.01, dipTotalGamma * stdDevSpread * fairValue)
+    : dipTotalGamma * stdDevSpread * fairValue;
+
+  console.log(scalper.symbol, 'Position Gamma Γ:', netGamma, 'Fair Value', fairValue);
+  if ((netGamma * fairValue) < (scalper.minSize * fairValue)) {
+    console.log(scalper.symbol, 'Gamma Hedge Too Small');
```

Comment:
log the hedge amount

Reply:
and move the log line above to below so we dont get redundant logs when this happens